src/common/operator/cast_operators.cpp

```diff
@@ -816,7 +816,7 @@ struct IntegerCastOperation {
 		if (dbl_res < NumericLimits<result_t>::Minimum() || dbl_res > NumericLimits<result_t>::Maximum()) {
 			return false;
 		}
-		state.result = (result_t)dbl_res;
+		state.result = (result_t)std::nearbyint(dbl_res);
```

Comment:
[`std::nearbyint`](https://en.cppreference.com/w/cpp/numeric/math/nearbyint) uses global state from `std::fesetround`, which could be potentially problematic when integrating DuckDB in various clients as clients might change this setting. Is there any alternative we could explore here?

Reply:
I'll have a look, but if it is part of the C++ runtime, I think we would be isolated.  Also, if the client wants a different setting, maybe it makes sense to let them override it?

Reply:
Looks like it is [a thread local processor setting](https://en.cppreference.com/w/cpp/numeric/fenv), but I think this is unusual enough that a if someone has changed it we should follow that setting. After all, we are an embedded acceleration engine, not a stand-alone server.